### PR TITLE
Migrate Convolution Tests from Boost to catch2

### PR DESCRIPTION
**Pull request by knightxerons**

### CODE REVIEW

File: src/mlpack/tests/convolution_test.cpp [hunk 1]
```diff
@@ -46,13 +45,13 @@ void Convolution2DMethodTest(const arma::mat input,
...
       (convOutput.n_cols == output.n_cols);
-  BOOST_REQUIRE_EQUAL(b, 1);
+  REQUIRE(b == 1);
 
   const double* outputPtr = output.memptr();
   const double* convOutputPtr = convOutput.memptr();
 
   for (size_t i = 0; i < output.n_elem; ++i, outputPtr++, convOutputPtr++)
-    BOOST_REQUIRE_CLOSE(*outputPtr, *convOutputPtr, 1e-3);
+    REQUIRE(*outputPtr == Approx(*convOutputPtr).epsilon(1e-3));
```
Comment (zoq) [inline src/mlpack/tests/convolution_test.cpp:54]: You need to divide by 100, see https://github.com/mlpack/mlpack/pull/2525#discussion_r456959546

Comment (knightxerons) [reply src/mlpack/tests/convolution_test.cpp:54]: Ohh, Thanks I'll Correct it.

**Comment by knightxerons** [comment]
Hi, I'm new here. Can someone please explain what I'm doing wrong? #2533

**Comment by jeffin143** [comment]
@knightxerons , can you rebase the pr against master, Let's see if this fails again

**Comment by zoq** [comment]
I resolved the merge conflict for you and also removed the `"ConvolutionalNetworkTest;" `. Make sure to run `git pull` before you commit anything new. Also I recommend to create a new branch if you open another PR.

**Comment by jeffin143** [comment]
Commenting : Linking the issue here #2523 , so that no one else parallely works on the same test

**Comment by knightxerons** [comment]
Thanks @jeffin143  and sure @zoq , I will look upto it in future.

File: src/mlpack/tests/convolution_test.cpp [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
-#include <boost/test/unit_test.hpp>
-#include "test_tools.hpp"
+#include "serialization_catch.hpp"
+#include "catch.hpp"
+#include "test_catch_tools.hpp"
 
 using namespace mlpack;
 using namespace mlpack::ann;
 
-BOOST_AUTO_TEST_SUITE(ConvolutionTest);
```
[the remark this answers is not in the document]
Comment (knightxerons) [reply src/mlpack/tests/convolution_test.cpp]: I'll do so, Thanks.

**Comment by knightxerons** [comment]
Glad to be part of this community!!

**Comment by knightxerons** [comment]
@birm , I'll keep that in mind and I'm definitely looking forward to contributing more. Thanks for your support!!

**Pull request merged**